src/lib/util/assert.ts

```diff
@@ -10,3 +10,14 @@ function assertPromise<T>(value: Promise<T>, message?: string): Promise<T> {
   assert(value instanceof Promise, message ?? 'Expected a promise');
   return value;
 }
+
+/**
+ * Assert that the value is not undefined, return the value.
+ */
+function assertDefined<T>(
```

Comment:
actually sad that type guards arent really useful in this scenario :(

Reply:
not sure if you meant that but this comment made me realized that I can remove `as T`!